docs/src/pages/error-decoder/ErrorDecoder.js

```diff
@@ -0,0 +1,67 @@
+import * as React from 'react';
+import { useRouter } from 'next/router';
+
+export default function ErrorDecoder() {
+  const {
+    query: { code, ...query },
+  } = useRouter();
+  const args = Array.isArray(query['args[]']) ? query['args[]'] : [query['args[]']];
+
+  const [data, dispatch] = React.useReducer(
+    (previousState, action) => {
+      switch (action.type) {
+        case 'rejected':
+          return { errorCodes: null, state: 'rejected' };
+        case 'resolved':
+          return { errorCodes: action.payload, state: 'resolved' };
+        default:
+          throw new Error(`We made a mistake passing an unknown action.`);
+      }
+    },
+    { errorCodes: null, state: 'loading' },
+  );
+
+  React.useEffect(() => {
+    let cancelled = false;
+
+    fetch('/static/error-codes.json')
+      .then((response) => {
+        return response.json();
+      })
+      .then((json) => {
+        if (cancelled === false) {
```

Comment:
Will it work for errors throwed outside of core? I am wondering because there are couple of errors that would be useful to minify in production.

Reply:
If this would be supported it's definitely a possibility for reducing wasteful fetches. I haven't looked at it because this approach works just as well? Is there some edge case I'm missing?